public/javascripts/stops.js

```diff
@@ -27,13 +27,33 @@ class Stop {
 			parseFloat(stopNode.getAttribute("lat")),
 			parseFloat(stopNode.getAttribute("lng"))
 		];
+		
+		this.selected = false;
+		this.zoomShown = false;
 
 		this.marker = L.circle(this.coords, {
 			radius: 5,
 			color: markerColors.unknown,
 			fillColor: markerColors.unknown,
 			fillOpacity: 1,
 		});
+
+		this.marker.on("click", e => {
+			for (let route of routesByOrder) {
+				route.selected = false;
+
+				route.updateShown();
+			}
+
+			// Hides any shown lines and any extra stops on the line. Also opens
+			// the sidebar tab for the clicked on stop
+			updateShownLines();
+			sidebarTabs.showTab("stops");
+			showStop(this);
+
+			//Centers the map on a given stop based on marker clicked
+			centerOnMarker(this.marker);
+		});
```

Comment:
I would prefer to have event listeners outside of the `Stop` and `Route` classes to ensure good model/GUI separation.  Could you register the event listeners in `stopsTable.js`?  `createStopButtons()` might be a good place.